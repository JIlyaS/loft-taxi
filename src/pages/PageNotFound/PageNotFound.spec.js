import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';

import {Provider} from 'react-redux';

import {PageNotFound} from './';
import createStore from '../../store';

const store = createStore();

const renderWithRouter = (component, {route='/', history = createMemoryHistory({ initialEntries: [route]})} = {}) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <Router history={history}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          { children }
        </MuiPickersUtilsProvider>
      </Router>
    </Provider>
  );

  return {
    ...render(component, {wrapper: Wrapper}),
    history,
  };
};


describe('PageNotFound', () => {
  it('renders correctly spanshot', () => {
    const history = createMemoryHistory();
    const tree = renderer.create(<Provider store={store}>
      <Router history={history}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <PageNotFound />
        </MuiPickersUtilsProvider>
      </Router>
    </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render correct', () => {
    const { queryByTestId } = renderWithRouter(<PageNotFound />);
    expect(queryByTestId('not-found')).toBeInTheDocument();
  });
});
