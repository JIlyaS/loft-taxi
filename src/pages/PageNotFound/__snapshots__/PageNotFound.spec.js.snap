// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders correctly spanshot 1`] = `
<main
  className="main-page"
  data-testid="not-found"
>
  Страница не найдена
</main>
`;
